src/hnsw.c

```diff
@@ -169,25 +170,62 @@ Datum       hnsw_handler(PG_FUNCTION_ARGS)
     PG_RETURN_POINTER(amroutine);
 }
 
-static float4 calc_distance(ArrayType *a, ArrayType *b)
+static float4 array_dist(ArrayType *a, ArrayType *b, usearch_metric_kind_t metric_kind)
 {
-    int     a_dim = ArrayGetNItems(ARR_NDIM(a), ARR_DIMS(a));
-    int     b_dim = ArrayGetNItems(ARR_NDIM(b), ARR_DIMS(b));
-    float4 *ax = (float4 *)ARR_DATA_PTR(a);
-    float4 *bx = (float4 *)ARR_DATA_PTR(b);
+    int a_dim = ArrayGetNItems(ARR_NDIM(a), ARR_DIMS(a));
+    int b_dim = ArrayGetNItems(ARR_NDIM(b), ARR_DIMS(b));
 
     if(a_dim != b_dim) {
         elog(ERROR, "expected equally sized arrays but got arrays with dimensions %d and %d", a_dim, b_dim);
     }
 
-    return l2sq_dist_impl(ax, bx, a_dim);
+    float4 *ax = (float4 *)ARR_DATA_PTR(a);
+    float4 *bx = (float4 *)ARR_DATA_PTR(b);
+
+    return usearch_dist(ax, bx, metric_kind, a_dim, usearch_scalar_f32_k);
+}
+
+static float8 vector_dist(Vector *a, Vector *b, usearch_metric_kind_t metric_kind)
+{
+    if(a->dim != b->dim) {
+        elog(ERROR, "expected equally sized vectors but got vecors with dimensions %d and %d", a->dim, b->dim);
+    }
+
+    return usearch_dist(a->x, b->x, metric_kind, a->dim, usearch_scalar_f32_k);
 }
 
+PGDLLEXPORT PG_FUNCTION_INFO_V1(ldb_generic_dist);
+Datum       ldb_generic_dist(PG_FUNCTION_ARGS) { PG_RETURN_NULL(); }
```

Comment:
Is this ever called? can this return an error instead to make sure it is never called directly?